contracts/TalentLayerService.sol

```diff
@@ -341,75 +457,129 @@ contract TalentLayerService is Initializable, ERC2771RecipientUpgradeable, UUPSU
     function createProposal(
         uint256 _profileId,
         uint256 _serviceId,
-        address _rateToken,
         uint256 _rateAmount,
         uint256 _platformId,
         string calldata _dataUri,
         uint256 _expirationDate,
         bytes calldata _signature
     ) public payable onlyOwnerOrDelegate(_profileId) {
-        _validateProposal(_profileId, _serviceId, _rateToken, _rateAmount, _platformId, _dataUri, _signature);
+        _validateProposal(_profileId, _serviceId, _rateAmount, _platformId, _dataUri, _signature, 0);
+
+        proposals[_serviceId][_profileId] = Proposal({
+            status: ProposalStatus.Pending,
+            ownerId: _profileId,
+            rateToken: address(0),
+            rateAmount: _rateAmount,
+            platformId: _platformId,
+            dataUri: _dataUri,
+            expirationDate: _expirationDate,
+            referrerId: 0
+        });
+
+        if (serviceNonce[_profileId] == 0 && proposalNonce[_profileId] == 0) {
+            tlId.setHasActivity(_profileId);
+        }
+        proposalNonce[_profileId]++;
+
+        emit ProposalCreatedWithoutToken(
+            _serviceId,
+            _profileId,
+            _dataUri,
+            ProposalStatus.Pending,
+            _rateAmount,
+            _platformId,
+            _expirationDate,
+            0
+        );
+    }
+
+    /**
+     * @notice Allows an seller to propose his service for a service
+     * @param _profileId The TalentLayer ID of the user owner of the proposal
+     * @param _serviceId The service linked to the new proposal
+     * @param _rateAmount the amount of token chosen
+     * @param _dataUri token Id to IPFS URI mapping
+     * @param _platformId platform ID from where the proposal is created
+     * @param _expirationDate the time before the proposal is automatically validated
+     * @param _signature optional platform signature to allow the operation
+     * @param _referrerId the id of the referrer (Zero if no referrer)
+     */
+    function createProposalWithReferrer(
+        uint256 _profileId,
+        uint256 _serviceId,
+        uint256 _rateAmount,
+        uint256 _platformId,
+        string calldata _dataUri,
+        uint256 _expirationDate,
+        bytes calldata _signature,
+        uint256 _referrerId
```

Comment:
- don't create a new function, and move the new param _referrerId to createProposal